crates/bevy_math/src/rects/rect.rs

```diff
@@ -277,6 +277,27 @@ impl Rect {
         r
     }
 
+    /// Check if this rectangle contains another rectangle.
+    /// Touching edges count as containing. An empty rectangle never contains anything.
+    ///  
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_math::Rect;
+    /// let r1 = Rect::new(0., 0., 5., 1.);
+    /// let r2 = Rect::new(1., -1., 3., 3.);
+    /// let r3 = Rect::new(6., 0., 1., 1.);
+    /// assert!(r1.contains_rect(r2));
+    /// assert!(r1.contains_rect(r3));
```

Comment:
```suggestion
    /// let r3 = Rect::new(6., 0., 1., 1.);
    /// let r4 = Rect::new(0., 0., 0., 0.);
    /// assert!(r1.contains_rect(r2));
    /// assert!(r1.contains_rect(r3));
    /// assert!(r4.is_empty() && r4.contains_rect(r4));
```

`r4` is empty, but contains itself

Reply:
I think we should add this to the doc test.